### PR TITLE
[#259] Add Yarn Support

### PR DESCRIPTION
**Pull request by RRorg**
A possible solution for #259

The following changes are contained in this pull request:
- update all Dev-Dependencies to their latest stable releases
- use yarn instead of npm for this project

- [X] `yarn run checks` script has been run without errors.

Implementing the yarn support like this is not a small step, since it *requires* contributors to use yarn. A "softer", more tolerant approach would be merely adding and maintaining the `yarn.lock` file, while still running all scripts via npm, thus making yarn optional.

Forcing contributors to use yarn is a potentially controversial step, so let's only do this if all current contributors are in favor of it. I've marked you as reviewers.

Please delete your `node_modules` folder when switching over from npm, and reinstall with yarn.

ninja edit: I didn't want to mess with the Travis settings, so let's not forget changing them when we switch over.

### CODE REVIEW

**Comment by RRorg** [comment]
I feel like Travis support for Yarn is just not yet there, at least, until it comes pre-installed on the build images. Let's just update the dependencies and add an optional Yarn lockfile.

**Comment by pmeinhardt** [comment]
Yarn should be there on Node.js images I think. Looking at the `.travis.yml` it seems like no `language: node_js` is present though… The default image is Ruby if I remember correctly. Adding this to the top of the config may be worth a try?

**Comment by RRorg** [comment]
Good call, I'll revisit this!

File: README.md [hunk 1]
```diff
@@ -190,9 +190,11 @@ Clone this repository, `cd` into the folder created, then install dependencies:
 ```sh
 git clone https://github.com/bitcrowd/bitstyles.git
 cd bitstyles
-npm install
+yarn
 ```
 
+It is recommended to use `yarn` instead of `npm install` to ensure version consistency across dependencies.
```
Comment (planktonic) [inline README.md]: Can you add a link to yarn here please? e.g. https://yarnpkg.com/en/docs/install

File: package.json [hunk 2]
```diff
@@ -9,21 +9,18 @@
     "lint:scss": "postcss -c postcss_lint.json",
     "lint:js": "eslint .",
     "lint": "npm run lint:scss && npm run lint:js",
-
```
Comment (planktonic) [inline package.json]: Was this deliberate or just automatic editor behaviour? I’d prefer to keep this section chunked as it’s so long; if necessary add blank comments.

Comment (RRorg) [inline package.json]: sorry, my bad, has to have been something automatic

File: package.json [hunk 3]
```diff
@@ -60,22 +57,22 @@
...
+    "cssnano": "^3.10.0",
+    "eslint": "^3.13.1",
     "eslint-config-bitcrowd-base": "^8.0.0",
-    "eslint-plugin-import": "^1.16.0",
-    "node-sass": "^3.10.1",
-    "parker": "1.0.0-alpha.0",
-    "postcss": "^5.2.4",
+    "eslint-plugin-import": "^2.2.0",
+    "node-sass": "^4.3.0",
+    "parker": "^0.0.10",
```
Comment (RRorg) [inline package.json]: reverted

**Comment by RRorg** [comment]
If you're all in favor, this can be merged now. Not full-on-yarn yet, but with added yarn support.

image

**Comment by planktonic** [comment]
Thanks, looks good!

**Pull request merged**